code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+	var/fusion_temperature = 0
+	var/moderator_temperature = 0
+	var/coolant_temperature = 0
+	var/output_temperature = 0
+	///Var used in the meltdown phase
+	var/final_countdown = FALSE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/Initialize()
+	. = ..()
+	internal_fusion = new
+	internal_fusion.assert_gases(/datum/gas/hydrogen, /datum/gas/tritium)
+	internal_output = new
+	moderator_internal = new
+
+	radio = new(src)
+	radio.keyslot = new radio_key
+	radio.listening = 0
+	radio.recalculateChannels()
+	investigate_log("has been created.", INVESTIGATE_HYPERTORUS)
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/SetInitDirections()
+	switch(dir)
+		if(NORTH, SOUTH)
+			initialize_directions = EAST|WEST
+		if(EAST, WEST)
+			initialize_directions = NORTH|SOUTH
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/Destroy()
+	if(linked_input)
+		QDEL_NULL(linked_input)
+	if(linked_output)
+		QDEL_NULL(linked_output)
+	if(linked_moderator)
+		QDEL_NULL(linked_moderator)
+	if(linked_interface)
+		QDEL_NULL(linked_interface)
+	if(corners.len)
+		for(var/corner in corners)
+			QDEL_NULL(corner)
+	QDEL_NULL(radio)
+	QDEL_NULL(soundloop)
+	return..()
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] can be rotated by first opening the panel with a screwdriver and then using a wrench on it.</span>"
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/update_icon()
+	if(panel_open)
+		icon_state = icon_state_open
+	else if(active)
+		icon_state = icon_state_active
+	else
+		icon_state = icon_state_off
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/getNodeConnects()
+	return list(turn(dir, 270), turn(dir, 90))
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/can_be_node(obj/machinery/atmospherics/target)
+	if(anchored)
+		return ..()
+	return FALSE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/attackby(obj/item/I, mob/user, params)
+	if(!fusion_started)
+		if(default_deconstruction_screwdriver(user, icon_state_open, icon_state_off, I))
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/default_change_direction_wrench(mob/user, obj/item/I)
+	. = ..()
+	if(!.)
+		return
+	if(!anchored)
+		return FALSE
+	var/obj/machinery/atmospherics/node1 = nodes[1]
+	var/obj/machinery/atmospherics/node2 = nodes[2]
+	if(node1)
+		node1.disconnect(src)
+		nodes[1] = null
+		nullifyPipenet(parents[1])
+	if(node2)
+		node2.disconnect(src)
+		nodes[2] = null
+		nullifyPipenet(parents[1])
+
+	SetInitDirections()
+	atmosinit()
+	node1 = nodes[1]
+	if(node1)
+		node1.atmosinit()
+		node1.addMember(src)
+	node2 = nodes[2]
+	if(node2)
+		node2.atmosinit()
+		node2.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
```

Comment:
I hate this, but there's nothing to be done

Reply:
same, but is either this, or fixed core